doc/classes/StringName.xml

```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="StringName" version="4.0">
 	<brief_description>
+		A type of string for representing names.
 	</brief_description>
 	<description>
+		[StringName]s are immutable strings designed for general-purpose representing of names. [StringName] ensures that only one instance of given name exists (so two [StringName]s with the same value are the same object). Comparing them is much faster than with regular [String]s.
```

Comment:
Perhaps it would be a good idea to mention that it uses pointer comparison under the hood, which is why comparison is faster.